core/src/components/datetime/datetime.tsx

```diff
@@ -1364,10 +1365,20 @@ export class Datetime implements ComponentInterface {
               hour: ev.detail.value
             });
 
+            if (minutesPickerColumnRef) {
+              raf(() => {
+                // When the time picker is re-rendered, the minutes columns does not
+                // scroll to the correct position. This is a workaround to ensure
+                // that the minutes column displays the correct value.
+                minutesPickerColumnRef!.scrollActiveItemIntoView();
```

Comment:
Are we able to solve this inside of `ion-picker-column-internal`? Seems to be specific to this component's rendering, not datetime.

Reply:
Tried solving from inside `ion-picker-column-internal`, but even though the watch callback for value is called on the re-render, the double state variable change (`activeParts` and `workingParts`), seems to cause issues. I tried also request animation frame wrapping the scroll active item within the picker and that also did not work.

It seems like the behavior happening is:
- Hour column is changed
- `workingParts` are updated 
  - re-render in progress
  - Assigns new value to `ion-picker-column-internal`
  - Scroll active item into view fires
 - `activeParts` are updated
   - re-render in progress
   - Conflicts with change from previous re-render / `ion-picker-column-internal` scroll position is reset

Open to alternatives here, the two state variables being updated in a single call is a problem in general that might require a larger discovery.

Reply:
Doing the following inside of `picker-column-internal` fixes the issue on my end:

```js
componentDidRender() {
    this.scrollActiveItemIntoView();
  }
```

We would likely want to do some additional checks here to make sure we don't do an excessive amount of scroll requests here.

This is the test case I used:

```html
<ion-datetime min="2022-02-24T10:15:00" value="2022-02-24T10:16:00"></ion-datetime>
```

1. Open time picker. Observe that it starts on 10:16.
2. Select 11 in the hour column. Observe that the time is 11:16.

-----

One thing we should probably figure out after we deal with the above is what to do if the value that you had selected is no longer available. If you were on 11:10 and then moved the hour to be 10. The minutes are now out of bounds -- should we select the minimum value?

Reply:
Here is a video of the fix in action:
https://user-images.githubusercontent.com/2721089/155559699-16a971df-7e4b-46b3-9b6e-6deb5893447a.mov

Reply:
This is awesome, thanks! I had not thought of hooking into the render lifecycle. I'm going to experiment with this and some other changes I was exploring around minute values no longer being available (as a result of min/max ranges). 

I'll need to think more on what the picker should do with values that are no longer available. For time selections, I think it could make sense that it scrolls the wheel to the zero position (whatever the minimum bounds is). For more generic options, such as a picker wheel of colors, defaulting to the zero position seems incorrect and a blank starting point would make sense. This could be controlled based on what options you render though (passing in a blank option if you don't want a default selection for out of bounds values). 

I'll also need to make sure that if a time value (i.e. minute) is unavailable and it scrolls the active item state to `00` for instance, that it would actually commit a value change. Otherwise users would incorrectly believe their minute value is `00`, even though the active value is something not visible. It would also require users to scroll to the next item and then to the previous, to lock in `00`.